pennylane/math/utils.py

```diff
@@ -157,7 +157,13 @@ def convert_like(tensor1, tensor2):
     >>> convert_like(x, y)
     <tf.Tensor: shape=(2,), dtype=int64, numpy=array([1, 2])>
     """
-    return np.asarray(tensor1, like=get_interface(tensor2))
+    interface = get_interface(tensor2)
+
+    if interface == "torch":
+        dev = tensor2.device
+        return np.asarray(tensor1, device=dev, like=interface)

```

Comment:
This did the trick! Everything stays on the GPU now and I am no longer getting errors when computing the gradients :tada:

Reply:
Yay! Does it work for finite-diff, adjoint forward, and adjoint backwards as well?

Reply:
Yeah! :100:

Reply:
🤯 

That's like when you modify your code and it compiles on first go!

Reply:
One of the initial tests I did was adjoint backwards actually, I didn't even realize! I just searched for `torch.tensor` in the test file and changed the device of a number of them.

Reply:
oh nice!!